### PR TITLE
More graph dumps, better named.

### PR DESCRIPTION
**Pull request by cconvey**
## Description ##
(Brief description on what this PR is about)

### Essentials ###
- [ ] Passed code style checking (`make lint`)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: src/ngraph/ngraph_utils.h [hunk 1]
```diff
@@ -29,6 +28,7 @@
 #include <vector>
 
 #include "ngraph_graph.h"
+#include "mxnet/ndarray.h"
```
Comment (mbrookhart) [inline src/ngraph/ngraph_utils.h]: I think this reorder will break `make cpplint`

Comment (cconvey) [reply src/ngraph/ngraph_utils.h]: Actually it does.  I try to always run `make cpplint` prior to creating a PR.  

Should we add `make cpplint` this to our CI pre-merge tests?

Comment (mbrookhart) [reply src/ngraph/ngraph_utils.h]: Yeah, we should.

File: src/ngraph/ngraph_sgcompiler_utils.h [hunk 2]
```diff
@@ -83,6 +83,9 @@ inline ngraph::Shape TShape_to_NShape(const nnvm::TShape& inshape) {
   return convert_shapes<nnvm::TShape, ngraph::Shape>(inshape);
 }
 
+// FIXME: Can we replace all of the 'makeConstant' machinery below with
+// calls to 'ngraph::op::Constant::create(...)'?
```
Comment (mbrookhart) [inline src/ngraph/ngraph_sgcompiler_utils.h]: We don't want to create large constants arrays in memory directly, the scalar->broadcast mechanism is still easier to optmize, and removing these utilities leads to a lot of code copy for extract shape/ extract data type / create constant/ broadcast

File: src/ngraph/ngraph_sgcompiler.cc [hunk 3]
```diff
@@ -58,7 +45,7 @@ void CompileForward(std::shared_ptr<Graph> sub_graph,
 
   // Log the graph so Graph_* corresponds to Function_* in codgen
   if (ngraph_log_graph) {
-    dump_graph(f);
+    dump_graph(f, __FUNCTION__, "fprop");
```
Comment (mbrookhart) [inline src/ngraph/ngraph_sgcompiler.cc]: What is `__FUNCTION__`?

**Pull request merged**